File: backend/app/schemas/simulator.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field

class SimulatorRequest(BaseModel):
    scenario_name: str = "Custom Stress Test"
    patient_surge_pct: float = Field(default=25.0, ge=0.0, le=200.0)
    supplier_delay_days: int = Field(default=3, ge=0, le=30)
    staff_absenteeism_pct: float = Field(default=10.0, ge=0.0, le=75.0)
    apply_ai_redistribution: bool = True
    apply_emergency_buffer: bool = True

class CriticalBottleneck(BaseModel):
    entity: str
    risk_type: str  # "STOCKOUT", "BED_CAPACITY", "STAFF_DEFICIT"
    days_until_breach: float
    severity: str  # "MODERATE", "CRITICAL", "CATASTROPHIC"
    description: str

class SimulatorResponse(BaseModel):
    scenario_name: str
    baseline_resilience: float
    simulated_resilience: float
    resilience_delta: float
    projected_bed_occupancy_pct: float
    stockout_risk_skus_count: int
    critical_bottlenecks: List[CriticalBottleneck]
    recommended_mitigations: List[str]
    simulated_at: str

File: frontend/src/types/simulator.ts
export interface CriticalBottleneck {
  entity: string
  risk_type: "STOCKOUT" | "BED_CAPACITY" | "STAFF_DEFICIT"
  days_until_breach: number
  severity: "MODERATE" | "CRITICAL" | "CATASTROPHIC"
  description: string
}

export interface SimulatorRequest {
  scenario_name: string
  patient_surge_pct: number
  supplier_delay_days: number
  staff_absenteeism_pct: number
  apply_ai_redistribution: boolean
  apply_emergency_buffer: boolean
}

export interface SimulatorResponse {
  scenario_name: string
  baseline_resilience: number
  simulated_resilience: number
  resilience_delta: number
  projected_bed_occupancy_pct: number
  stockout_risk_skus_count: number
  critical_bottlenecks: CriticalBottleneck[]
  recommended_mitigations: string[]
  simulated_at: string
}
